### PR TITLE
documentation: refactor registers and function signatures

### PR DESCRIPTION
**Pull request by carenas**
Slightly expanded to avoid confusing references and describe better the way that the use of scratch registers affect `sljit_emit_enter()`

Fixes: #168

### CODE REVIEW

File: sljit_src/sljitLir.h [hunk 1]
```diff
@@ -258,20 +258,32 @@ extern "C" {
...
+
+   Furthermore, the index that will be used for the scratch register depends
+   on how many previous integer registers (scratch or saved) were allocated,
+   starting with SLJIT_R0. See below for some examples.
+
+   The floating point arguments always use scratch registers (but not the
+   _R suffix like in the integer registers) and must be placed into SLJIT_FR0,
+   SLJIT_FR1, just like in the "call" functions.
 
    Note: if a function is called by sljit_emit_call/sljit_emit_icall and
```
Comment (carenas) [inline sljit_src/sljitLir.h:288]: I don't understand this "Note", specially the references to the index affecting the floating point register index somehow and that would seem to conflict with the fact that floating point registers should be always using `SLJIT_FR0 ..`

File: sljit_src/sljitLir.h [hunk 2]
```diff
@@ -127,40 +127,40 @@ extern "C" {
...
   The purpose of this definition is convenience: saved registers can
-  be used as extra scratch registers. For example four registers can
-  be specified as scratch registers and the fifth one as saved register
-  on the CPU above and any user code which requires four scratch
-  registers can run unmodified. The SLJIT compiler automatically saves
-  the content of the two extra scratch register on the stack. Scratch
-  registers can also be preserved by saving their value on the stack
-  but this needs to be done manually.
+  be used as extra scratch registers. For example, in the previous
+  architecture four registers can be specified as scratch registers
```
Comment (carenas) [inline sljit_src/sljitLir.h:148]: s/architecture four/sample CPU, four/

the rest of the paragraph also needs more commas

File: sljit_src/sljitLir.h (hunk 1 above)
Comment (zherczeg) [reply sljit_src/sljitLir.h:288]: It wants to explain that floating point arguments cannot be moved to saved registers, so the _R prefix is meaningless, and not used.

Thank you for improving the text.

File: sljit_src/sljitLir.h (hunk 2 above)
Comment (zherczeg) [reply sljit_src/sljitLir.h:148]: Ok. You can also rephrase them to use more sentences.

File: sljit_src/sljitLir.h (hunk 1 above)
Comment (carenas) [reply sljit_src/sljitLir.h:288]: I was referring to the note in https://github.com/zherczeg/sljit/blame/47ed75d0bcbe77f3e82b89e6e27f6b535815b821/sljit_src/sljitLir.h#LL276-L279C47

Comment (zherczeg) [reply sljit_src/sljitLir.h:288]: The note explains that if an argument is not moved to a saved register, the register assigned to the argument "keeps" its value. I.e., if you put an argument to R2 for the call, it is still in R2 after emit_enter, unless it is moved to a saved register (R2 is undefined in the latter case, just like the other registers).

**Comment by zherczeg** [comment]
If this is the final version, I can land it

**Review by zherczeg — APPROVED** [review]
LGTM

**Pull request merged**